docs/table-partitioning-bucketing-sort-order.md

```diff
@@ -0,0 +1,296 @@
+| field           | `{"type":"field","fieldName":["score"]}`                          | `FieldReferenceDTO.of("score")`                                                                         | `score`                  | field reference value `score`  |
+| function        | `{"type":"function","functionName":"hour","fieldName":["score"]}` | `new FuncExpressionDTO.Builder()<br/>.withFunctionName("hour")<br/>.withFunctionArgs("score").build()`  | `hour(score)`            | function value `hour(score)`   |
+| constant        | `{"type":"literal","value":10, "dataType": "integer"}`            | `new LiteralDTO.Builder()<br/>.withValue("10")<br/>.withDataType(Types.IntegerType.get())<br/>.build()` | `10`                     | Integer constant `10`          |
+
+
+<Tabs>
+<TabItem value="Json" label="Json">
+
+```json
+{
+  "strategy": "hash",
+  "number": 4,
+  "funcArgs": [
+    {
+      "type": "field",
+      "fieldName": ["score"]
+    }
+  ]
+}
+```
+
+</TabItem>
+<TabItem value="java" label="Java">
+
+```java
+ new DistributionDTO.Builder()
+    .withStrategy(Strategy.HASH)
+    .withNumber(4)
+    .withArgs(FieldReferenceDTO.of("score"))
+    .build()
+```
+
+</TabItem>
+</Tabs>
+
+
+## Sorted order table
+
+To define a sorted order table, you should use the following three components to construct a valid sorted order table.
+
+- Direction. It defines in which direction we sort the table. The default value is `ascending`.
+
+| Direction  | Description                               | Json   | Java                       |
+|------------|-------------------------------------------| ------ | -------------------------- |
+| ascending  | Sorted by a field or a function ascending | `asc`  | `SortDirection.ASCENDING`  |
+| descending | Sorted by a field or a function descending| `desc` | `SortDirection.DESCENDING` |
+
+- Null ordering. It describes how to handle null value when ordering
+
+| Null ordering Type | Description                       | Json          | Java                       |
+|--------------------|-----------------------------------| ------------- | -------------------------- |
+| null_first         | Put null value in the first place | `nulls_first` | `NullOrdering.NULLS_FIRST` |
+| null_last          | Put null value in the last place  | `nulls_last`  | `NullOrdering.NULLS_LAST`  |
+
+Note: If the direction value is `ascending`, the default ordering value is `nulls_first` and if the direction value is `descending`, the default ordering value is `nulls_last`.
+
+- Sort term. It shows which field or function should be used to sort the table, please refer to the `Expression type` in the bucketed table chapter.
+
+<Tabs>
+<TabItem value="Json" label="Json">
+
+```json
+ {
+  "direction": "asc",
+  "nullOrder": "NULLS_LAST",
+  "sortTerm":  {
+    "type": "field",
+    "fieldName": ["score"]
+  }
+}
+```
+
+</TabItem>
+<TabItem value="java" label="Java">
+
+```java
+ new SortOrderDTO.Builder()
```

Comment:
The Java demo should use the public static method in `com.datastrato.gravitino.rel.expressions.sorts.SortOrders` of API module?

Reply:
OK, I change the code here.